### PR TITLE
Upgrade to tokio v1.0

### LINKED ISSUE #1329: Upgrade to tokio v1.0
https://tokio.rs/blog/2020-12-tokio-1-0

### PR DESCRIPTION
**Pull request by messense**
Closes #1329

### CODE REVIEW

**Review by djc — APPROVED** [review]
LGTM, I'll leave it to @bluejekyll to look at before merging.

**Comment by zonyitoo** [comment]
This PR works well in my project. Looking forward to be merged soon. @bluejekyll

**Comment by bluejekyll** [comment]
Lots of festivities the last couple of days. I’ll review this today.

File: crates/openssl/src/tls_stream.rs
Comment (bluejekyll) [inline crates/openssl/src/tls_stream.rs:139]: Any reason we’re unwrapping rather than returning the error here?

File: crates/server/src/server/server_future.rs
Comment (bluejekyll) [inline crates/server/src/server/server_future.rs:242]: Another unwrap?

**Review by bluejekyll — CHANGES_REQUESTED** [review]
There are a couple of unwraps in what looks like non-test code. Can we return errors instead in those locations?

File: crates/openssl/src/tls_stream.rs
Comment (messense) [inline crates/openssl/src/tls_stream.rs:141]: @bluejekyll  I've removed the `unwrap`, but I am not sure what is the best error kind to propagate.

Comment (bluejekyll) [reply crates/openssl/src/tls_stream.rs:141]: This is fine for now. Thanks for making the change. If it occurs, we can make an explicit error type.

**Review by bluejekyll — APPROVED** [review]
Thanks for this PR! I just want to double check some things in the morning, but think it can be merged now.

**Comment by bluejekyll** [comment]
Merging, Thanks again for the PR!

**Pull request merged**
